apple/internal/partials/app_assets_validation.bzl

```diff
@@ -61,16 +61,25 @@ def _app_assets_validation_partial_impl(
                 "*.stickersequence)"
             )
             bundling_support.ensure_path_format(
-                "app_icons",
-                app_icons,
-                path_fragments,
+                attr = "app_icons",
+                files = app_icons,
+                path_fragments_list = path_fragments,
                 message = message,
             )
         elif platform_prerequisites.platform_type == apple_common.platform_type.tvos:
             bundling_support.ensure_single_xcassets_type(
-                "app_icons",
-                app_icons,
-                "brandassets",
+                attr = "app_icons",
+                extension = "brandassets",
+                files = app_icons,
+            )
+        elif platform_prerequisites.platform_type == apple_common.platform_type.visionos:
+            message = ("visionOS apps must use visionOS app icon layers grouped in " +
+                       ".solidimagestack bundles, not traditional App Icon Sets")
+            bundling_support.ensure_single_xcassets_type(
+                attr = "app_icons",
+                extension = "solidimagestack",
+                files = app_icons,
+                message = message,
             )
         elif platform_prerequisites.platform_type == getattr(apple_common.platform_type, "visionos", None):
```

Comment:
should remove this one if it's no longer needed

Reply:
Good catch, I missed it. Fixed now!